pkg/csi/service/vanilla/controller.go

```diff
@@ -518,12 +518,41 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 			}
 
 			// Get shared accessible datastores for matching topology requirement.
-			sharedDatastores, err = c.topologyMgr.GetSharedDatastoresInTopology(ctx,
-				commoncotypes.VanillaTopologyFetchDSParams{TopologyRequirement: topologyRequirement})
-			if err != nil || len(sharedDatastores) == 0 {
-				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to get shared datastores for topology requirement: %+v. Error: %+v",
-					topologyRequirement, err)
+			if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TopologyPreferentialDatastores) {
+				vcenter, err := c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
+				if err != nil {
+					return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+						"failed to get vCenter. Err: %v", err)
+				}
+				// If storage policy name is mentioned in storage class, retrieve the policy ID.
+				var storagePolicyID string
+				if scParams.StoragePolicyName != "" {
```

Comment:
move this logic inside `GetSharedDatastoresInTopology` and supply Policy name in the `VanillaTopologyFetchDSParams`

Reply:
Done